naptime/src/main/scala/org/coursera/naptime/errors.scala

```diff
@@ -35,8 +35,9 @@ case class NaptimeActionException(
     httpCode: Int,
     errorCode: Option[String],
     message: Option[String],
-    details: Option[JsValue] = None)
-    extends RuntimeException(s"Naptime error $httpCode [$errorCode]: $message") {
+    details: Option[JsValue] = None,
+    cause: Option[Throwable] = None)
+    extends RuntimeException(s"Naptime error $httpCode [$errorCode]: $message", cause.orNull) {
```

Comment:
should the responsibility of turning `cause` to a reasonable stack trace live in this case class as well?